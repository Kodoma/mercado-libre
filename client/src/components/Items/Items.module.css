.paper {
    padding: 5%;
    margin: 0 auto;
    display: flex;
    min-width: fit-content;
    border: none;
    boxShadow: none;
}

.custom {
    border: none;
    boxShadow: none;
}

.image {
    outline: none;
    width: 160px;
}

.img {
    object-fit: contain;
    display: block;
    aspect-ratio: auto 160 / 160;
}

.priceContainer {
    display: flex;
    align-content: center;
    align-items: center;
}

.freeShipping {
    margin-left: 10px;
}

.price,
.price:hover,
.price:link,
.price:visited,
.price:active,
.price:hover {
    color: #333333;
    text-decoration: none !important;
    outline: none !important;
}

.title,
.title:hover,
.title:link,
.title:visited,
.title:active,
.title:hover {
    color: #333333 !important;
    text-decoration: none !important;
    outline: none !important;
}

.condition {
    color: #878787
}

@media (max-width: 812px) {
    .paper {
        width: 100%;
        min-width: auto;
    }

}
