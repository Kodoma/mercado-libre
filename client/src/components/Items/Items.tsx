import React from "react";
import fetch from "cross-fetch";
import { useRouter } from 'next/router'
import Link from 'next/link'
import styles from './Items.module.css';
import {Box, Card, CardContent, CircularProgress, Grid, Paper, Typography} from "@material-ui/core";
import {Item} from "../../models/items";
import Price from "./Price/Price";
import Title from "./Title/Title";
import Breadcrumb from "../BreadCrumb/Breadcrumb";
import {isMobile} from "react-device-detect";
import Loading from "../Loading/Loading";


const ItemsList = () => {
    const [items, setItems] = React.useState(ItemsInitState);
    const router = useRouter();
    const search = (router) && router.query.search

    React.useEffect(() => {
        async function fetchItems(search: string) {
            if(!search) return;
            const request = await fetch(
                "http://0.0.0.0:8080/api/items?search="+search
            );
            const response = await request.json();
            setItems(response.items);
        }

        fetchItems(search as string);
    },[search]);

    const content = items && items.map( i => (
        <Card
            className={styles.custom}
            square={true}
            key={i.id}
        >
            <CardContent>
                <Paper
                    elevation={0}
                    className={styles.paper}>
                    <Grid container
                          spacing={2}
                          justify="center"
                          alignContent="center"
                          alignItems={(isMobile) ? "center": "flex-start"}
                          direction={(isMobile) ? "column": "row"}
                    >
                        <Grid item>
                            <Link
                                href={{
                                    pathname: `/items/${i.id}`,
                                }}
                            >
                                <img width={160} height={160} className={styles.img} src={i.thumbnail}  alt={i.title}/>
                            </Link>
                        </Grid>
                        <Grid item xs={12} sm container>
                            <Grid item xs container direction="column" spacing={2}>
                                <Grid item xs>
                                    <Price id={i.id} amount={i.price.amount} currency={i.price.currency} free_shipping={i.free_shipping}/>
                                    <Title id={i.id} title={i.title} />
                                </Grid>
                            </Grid>
                            <Grid item>
                                <Typography variant="subtitle1" className={styles.condition}>
                                    { i.condition }
                                </Typography>
                            </Grid>
                        </Grid>
                    </Grid>
                </Paper>
            </CardContent>
        </Card>
    ))

    return(
        <Box width={(isMobile) ? "100%" : "75%"}>
            <Breadcrumb search={search as string} />
            { (content.length === 0) ? <Loading /> : content }
        </Box>
    );
}

export default ItemsList

const ItemsInitState: Item[] = [];
