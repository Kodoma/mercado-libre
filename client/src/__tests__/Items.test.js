import React from 'react';
import 'jsdom-global/register';
import {mount, shallow} from 'enzyme';
import Items from '../components/Items/Items';

const useRouter = jest.spyOn(require('next/router'), 'useRouter')

describe('tests Items', () => {
    it('should render my Items component', () => {
        const component = shallow(<Items />);
        expect(component.getElements()).toMatchSnapshot();
    });

    it('should render Items', function () {
        useRouter.mockImplementationOnce(() => ({
            query: { search: 'iphone' },
        }))

        const wrap = mount(<Items />)

        expect(wrap.find('.custom')).toBeDefined()

    })
});
