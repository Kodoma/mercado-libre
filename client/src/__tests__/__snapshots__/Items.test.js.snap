// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`__tests__ Items should render my Items component 1`] = `
Array [
  <React.Fragment>
    <Styled(MuiBox)
      width="75%"
    >
      <EmptySearch />
    </Styled(MuiBox)>
  </React.Fragment>,
]
`;
